xml/telnet_banners.xml

```diff
@@ -816,4 +816,680 @@
+    <param pos="0" name="hw.family" value="OpenBSD"/>
+    <param pos="0" name="hw.product" value="OpenBSD"/>
+    <param pos="1" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="FreeBSD\/([^\\s]+)\s+\(([^\s]+)\)">
+    <description>a FreeBSD</description>
+    <!-- FreeBSD/amd64 (ms.gymspgs.cz) (pts/0)\n\n\n\nlogin: -->
+    <example _encoding="base64" os.arch="amd64" host.name="ms.gymspgs.cz">
+      RnJlZUJTRC9hbWQ2NCAobXMuZ3ltc3Bncy5jeikgKHB0cy8wKQoKCgpsb2dpbjo=
+    </example>
+    <param pos="0" name="hw.vendor" value="FreeBSD"/>
+    <param pos="0" name="hw.family" value="FreeBSD"/>
+    <param pos="0" name="hw.product" value="FreeBSD"/>
+    <param pos="1" name="os.arch"/>
+    <param pos="2" name="host.name"/>
+  </fingerprint>
+  <fingerprint pattern="NetBSD">
+    <description>NetBSD</description>
+    <!-- NetBSD/evbsh3 (Fukuyama.Host_AKS_0555_WL-v2.60d) (ttyp1)  -->
+    <example _encoding="base64">
+      TmV0QlNEL21lc3NpbWlwcyAoKSAodHR5cDMpCgpsb2dpbjog
+    </example>
+    <param pos="0" name="hw.vendor" value="NetBSD"/>
+    <param pos="0" name="hw.family" value="NetBSD"/>
+    <param pos="0" name="hw.product" value="NetBSD"/>
+  </fingerprint>
+  <fingerprint pattern="IRIX\W\((.*)\)">
+    <description>SGI IRIX</description>
+    <!-- IRIX (artemis.biol.uoa.gr)\n\n\n\nlogin: -->
+    <example _encoding="base64" host.name="artemis.biol.uoa.gr">
+      SVJJWCAoYXJ0ZW1pcy5iaW9sLnVvYS5ncikKCgoKbG9naW46IA==
+    </example>
+    <param pos="0" name="hw.vendor" value="SGI"/>
+    <param pos="0" name="hw.family" value="IRIX"/>
+    <param pos="0" name="hw.product" value="IRIX"/>
+    <param pos="1" name="host.name"/>
+  </fingerprint>
+  <fingerprint pattern="^(?m)(RS [^\s]+) System Software, Version ([^\s]+).*">
+    <description>RiverStone router</description>
+    <!-- ++++++++++++++++++++++++++++++++++++++++\nRS 8000 System Software, Version 9.3.0.5\n
+    Riverstone Networks, Inc., Copyright (c) 2000-2004. All rights reserved.\nSystem started on 2018-10-11 22:02:03
+    \n++++++++++++++++++++++++++++++++++++++++ -->
+    <example _encoding="base64" hw.product="RS 8000" os.version="9.3.0.5">
+      LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tL
+      S0tLS0tLQpSUyA4MDAwIFN5c3RlbSBTb2Z0d2FyZSwgVmVyc2lvbiA5LjMuMC41ClJpdmVyc3RvbmUgTmV0d2
+      9ya3MsIEluYy4sIENvcHlyaWdodCAoYykgMjAwMC0yMDA0LiBBbGwgcmlnaHRzIHJlc2VydmVkLgpTeXN0ZW0
+      gc3RhcnRlZCBvbiAyMDE4LTEwLTExIDIyOjAyOjAzCi0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0t
+      LS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS0tLS4uLg==
+    </example>
+    <param pos="0" name="hw.vendor" value="Riverstone"/>
+    <param pos="0" name="hw.family" value="Riverstone"/>
+    <param pos="0" name="hw.device" value="Router"/>
+    <param pos="1" name="hw.product"/>
+    <param pos="2" name="os.version"/>
+  </fingerprint>
+  <fingerprint pattern="(ES [^\s]+) System Software, Version ([^\s]+)">
+    <description>a Riverstone router`</description>
```

Comment:
trailing `.

Also, the RS and ES fingerprints could be combined, extracting the family.